### ISSUE TITLE
File in /tmp opened without O_EXCL

### ISSUE THREAD

**Issue by jcowgill** [gerbera/gerbera#157]
Gerbera opens a file in `/tmp` (the default value of `CFG_SERVER_TMPDIR`) without `O_EXCL` here (and later on in that function): https://github.com/gerbera/gerbera/blob/master/src/process.cc#L62

This is vulnerable to the classical `/tmp` directory race conditions (such as symlink racing) which may happen if the chosen filename is created by another user just before it is opened. This can be fixed by either opening the file using `O_EXCL`, or preferably using a function like `mkstemp`.

There are other uses of `tempName` in `src/transcoding/transcode_ext_handler.cc`, but I think they're safer because `mkfifo` will fail if the file exists.

**Comment by elmodaddyb** [comment]
@jcowgill   I took a look at the `open()` specification. Following your suggestion, I think the command should look something like below. 
```
    fd = open(input_file.c_str(), O_RDWR | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR);
```

I will have to try it out locally to hit this exact code. 
Any thought or suggestions?

**Comment by jcowgill** [comment]
Yes I think that would fix the bug.

**Comment by v00d00** [comment]
@elmodaddyb how is your patch testing going? Ready to merge?

**Comment by elmodaddyb** [comment]
I will likely test this this wknd, but first plan to solve the systemd cmake install PR.

**Comment by elmodaddyb** [comment]
@v00d00 @jcowgill 
I was able to make the recommended changes.  
I added a bogus **object.item.activeItem** which launched a script.  File access to `/tmp` seems to work as expected when using the `O_EXCL` flag
```
[src/process.cc:96] run_simple_process(): running /action/echoText.sh run < /tmp//mt_in_3Q8Y6Y > /tmp//mt_out_QS8Y6Y
```
Both input/output files were created as expected, and cleaned up after the process completed.
I am not sure how to test the race condition mentioned...but this seems like a good fix for now.

Let me know what you think, and I'll submit for next PR soon.

Thanks,
E
